### PR TITLE
feat(reputation): punish non revealer witnesses

### LINKED ISSUE #870: Implement non-revealears slashing

### PR DESCRIPTION
**Pull request by lrubiorod**
Close #870

### CODE REVIEW

File: node/src/actors/chain_manager/mod.rs [hunk 1]
```diff
@@ -466,18 +466,25 @@ impl ReputationInfo {
...
+                .map(|reveal_tx| {
+                    if true_revealer(reveal_tx, tally) {
+                        0
+                    } else {
+                        1
+                    }
+                })
+                .unwrap_or(1);
+            // Insert all the commiters, and increment their lie count by 1 if they don't reveal or
+            // if they lied (not reveal is the same as lie)
```
Comment (aesedepece) [inline node/src/actors/chain_manager/mod.rs]:
```suggestion
            // if they lied (withholding a reveal is treated the same as lying)
```

File: node/src/actors/chain_manager/mod.rs [hunk 2]
```diff
@@ -466,18 +466,25 @@ impl ReputationInfo {
...
+                .get(&pkh)
+                .map(|reveal_tx| {
+                    if true_revealer(reveal_tx, tally) {
+                        0
+                    } else {
+                        1
+                    }
+                })
+                .unwrap_or(1);
+            // Insert all the commiters, and increment their lie count by 1 if they don't reveal or
```
Comment (aesedepece) [inline node/src/actors/chain_manager/mod.rs]:
```suggestion
            // Insert all the committers, and increment their lie count by 1 if they fail to reveal or
```

**Comment by tmpolaczyk** [comment]
Tests?

File: node/src/actors/chain_manager/mod.rs [hunk 3]
```diff
@@ -769,3 +776,66 @@ fn show_info_dr(data_request_pool: &DataRequestPool, block: &Block) {
         );
     }
 }
+
+#[test]
```
Comment (tmpolaczyk) [inline node/src/actors/chain_manager/mod.rs]: Can you move this test to a `tests` module?

```rust
#[cfg(test)]
mod tests {
    pub use super::*;

```

**Pull request merged**